pkg/scheduler/server/api.go

```diff
@@ -169,6 +154,37 @@ func (s *Server) WatchJobs(stream schedulerv1pb.Scheduler_WatchJobsServer) error
 	}
 }
 
+func (s *Server) triggerJob(ctx context.Context, req *api.TriggerRequest) bool {
+	log.Debugf("Triggering job")
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*45)
+	defer cancel()
+
+	var meta schedulerv1pb.ScheduleJobMetadata
+	if err := req.GetMetadata().UnmarshalTo(&meta); err != nil {
+		log.Errorf("Error unmarshalling metadata: %s", err)
+		return true
+	}
+
+	idx := strings.LastIndex(req.GetName(), "||")
+	if idx == -1 || len(req.GetName()) <= idx+2 {
+		log.Errorf("Job name is malformed: %s", req.GetName())
```

Comment:
In this case, even if the job name is malformed, can we still send it back if the rest of the bits line up ok?

Reply:
No- this means that the actual key in ETCD is malformed so we can't do anythinghere.